src/action/quickbooks.action.ts

```diff
@@ -1,15 +1,27 @@
 import { QBTokenSelectSchemaType } from '@/db/schema/qbTokens'
-import { getPortalConnection } from '@/db/service/token.service'
+import {
+  getPortalConnection,
+  getSyncedPortalConnection,
+} from '@/db/service/token.service'
 
 export async function checkPortalConnection(
   portalId: string,
 ): Promise<QBTokenSelectSchemaType | null> {
   try {
-    const portalQBToken = await getPortalConnection(portalId)
-
-    return portalQBToken
+    return await getPortalConnection(portalId)
   } catch (err) {
-    console.error('getTokenPayload | Error =', err)
+    console.error('checkPortalConnection#getPortalConnection | Error =', err)
     return null
   }
 }
+
+export async function checkSyncStatus(portalId: string): Promise<boolean> {
+  try {
+    const syncedPortal: QBTokenSelectSchemaType | null =
+      await getSyncedPortalConnection(portalId)
+    return !syncedPortal ? false : syncedPortal?.syncFlag || false
```

Comment:
```suggestion
    const syncedPortal = await getSyncedPortalConnection(portalId)
    return syncedPortal?.syncFlag ?? false
```
Should be the same(?)